### DISCUSSION TITLE
osgEarth crashes on start up

### DISCUSSION THREAD

**Discussion by bclay1297** [pelicanmapping/osgearth#2630] [Support Q&A]
I am able to build osgEarth thanks to Glenn's help but now it crashes on startup.  I have tried multiple samples with the same results.
The crash is before it gets to the main of the test app. always in osgEarth.dll.  It seems to be related to OSG_INIT_SINGLETON_PROXY(osgEarthFileUtilsGetFullPathInitSingletonProxy, osgEarth::Util::getFullPath(std::string(), std::string()));

When it gets to osgEarth::Util::getFullPath(const std::string& relativeTo, const std::string &relativePath) both variables are empty strings

when it gets to osgEarth::Util::isRelativePath(const std::string& fileName) filename is empty so native is empty and indexing into an empty string causes the crash

Stack trace is shown below
>	osgEarthd.dll!osgEarth::Util::isRelativePath(const std::string & fileName) Line 171	C++
 	osgEarthd.dll!osgEarth::Util::getFullPath(const std::string & relativeTo, const std::string & relativePath) Line 264	C++
 	osgEarthd.dll!osgEarthFileUtilsGetFullPathInitSingletonProxy::osgEarthFileUtilsGetFullPathInitSingletonProxy() Line 379	C++
 	osgEarthd.dll!`dynamic initializer for 's_osgEarthFileUtilsGetFullPathInitSingletonProxy''() Line 379	C++

Bruce

**Comment by gwaldron** [comment]
I have never seen this or seen it reported before.
Do you have the same issue in a `RelWithDebInfo` configuration?

**Reply by bclay1297** [reply]
Glenn:
  
Thanks for your reply / help.

The RelWithDebug and release works better.  They have one startup error ... procedure entry point zip_close could not be located ... osgdb_zip.dll.

The program does come up and display the globe with the warnings shown below.  The font file mentioned does exist.  grid2.png is in C:\install\data.  I do not see what is causing the XML parsing warnings.  Some of the apps shut down when I zoom in without explanation.

C:\install\bin>osgearth_map --screen 1 0  --window 1 0 1920 1080 simple.earth
Fontconfig error: Cannot load default config file: No such file: (null)
Error reading file C:/WINDOWS/fonts\CascadiaCode.ttf: file not handled
Error reading file ../data/grid2.png: file not handled
[osgearth warning] XML parsing error
[osgearth warning] Error document empty. (row 0, col 0)
[osgearth warning]  1   Content-Type=image/png
[osgearth warning] [WMS] : Service Exception: Content-Type=image/png (URI=http://mesonet.agron.iastate.edu/cgi-bin/wms/nexrad/n0r.cgi?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=nexrad-n0r&FORMAT=image/png&STYLES=&SRS=EPSG:4326&WIDTH=256&HEIGHT=256&TRANSPARENT=TRUE&BBOX=-180.000000,-90.000000,0.000000,90.000000)
[osgearth warning] XML parsing error
[osgearth warning] Error document empty. (row 0, col 0)
[osgearth warning]  1   Content-Type=image/png
[osgearth warning] [WMS] : Service Exception: Content-Type=image/png (URI=http://mesonet.agron.iastate.edu/cgi-bin/wms/nexrad/n0r.cgi?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=nexrad-n0r&FORMAT=image/png&STYLES=&SRS=EPSG:4326&WIDTH=256&HEIGHT=256&TRANSPARENT=TRUE&BBOX=-90.000000,0.000000,0.000000,90.000000)
[osgearth warning] XML parsing error
[osgearth warning] Error document empty. (row 0, col 0)
[osgearth warning]  1   Content-Type=image/png
[osgearth warning] [WMS] : Service Exception: Content-Type=image/png (URI=http://mesonet.agron.iastate.edu/cgi-bin/wms/nexrad/n0r.cgi?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=nexrad-n0r&FORMAT=image/png&STYLES=&SRS=EPSG:4326&WIDTH=256&HEIGHT=256&TRANSPARENT=TRUE&BBOX=-180.000000,0.000000,-90.000000,90.000000)
[osgearth warning] XML parsing error
[osgearth warning] Error document empty. (row 0, col 0)
[osgearth warning]  1   Content-Type=image/png
[osgearth warning] [WMS] : Service Exception: Content-Type=image/png (URI=http://mesonet.agron.iastate.edu/cgi-bin/wms/nexrad/n0r.cgi?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=nexrad-n0r&FORMAT=image/png&STYLES=&SRS=EPSG:4326&WIDTH=256&HEIGHT=256&TRANSPARENT=TRUE&BBOX=-90.000000,0.000000,-45.000000,45.000000)
[osgearth warning] XML parsing error
[osgearth warning] Error document empty. (row 0, col 0)
[osgearth warning]  1   Content-Type=image/png
[osgearth warning] [WMS] : Service Exception: Content-Type=image/png (URI=http://mesonet.agron.iastate.edu/cgi-bin/wms/nexrad/n0r.cgi?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=nexrad-n0r&FORMAT=image/png&STYLES=&SRS=EPSG:4326&WIDTH=256&HEIGHT=256&TRANSPARENT=TRUE&BBOX=-135.000000,0.000000,-90.000000,45.000000)
[osgearth warning] XML parsing error
[osgearth warning] Error document empty. (row 0, col 0)
[osgearth warning]  1   Content-Type=image/png
[osgearth warning] [WMS] : Service Exception: Content-Type=image/png (URI=http://mesonet.agron.iastate.edu/cgi-bin/wms/nexrad/n0r.cgi?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=nexrad-n0r&FORMAT=image/png&STYLES=&SRS=EPSG:4326&WIDTH=256&HEIGHT=256&TRANSPARENT=TRUE&BBOX=-90.000000,45.000000,-45.000000,90.000000)
[osgearth warning] XML parsing error
[osgearth warning] Error document empty. (row 0, col 0)
[osgearth warning]  1   Content-Type=image/png
[osgearth warning] [WMS] : Service Exception: Content-Type=image/png (URI=http://mesonet.agron.iastate.edu/cgi-bin/wms/nexrad/n0r.cgi?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetMap&LAYERS=nexrad-n0r&FORMAT=image/png&STYLES=&SRS=EPSG:4326&WIDTH=256&HEIGHT=256&TRANSPARENT=TRUE&BBOX=-135.000000,45.000000,-90.000000,90.000000)

**Comment by gwaldron** [comment]
Some server in the map example is just not responding.
Start with the basics instead: `osgearth_imgui` or `osgearth_viewer`.

**Reply by bclay1297** [reply]
I was able to build and run release (non-debug) examples as well.  The results are not always what I would expect.  I pushed a list of the ones I tested and the results

[osgEarth_testResults.txt](https://github.com/user-attachments/files/17104403/osgEarth_testResults.txt)

I went back to the debug version and ran into problems again with rocksdb.lib. errors like

rocksdb.lib(hash.cc.obj) : error LNK2038: mismatch detected for 'RuntimeLibrary': value 'MD_DynamicRelease' doesn't match value 'MDd_DynamicDebug' in RocksDBCache.obj
4>rocksdb.lib(sharded_cache.cc.obj) : error LNK2038: mismatch detected for '_ITERATOR_DEBUG_LEVEL': value '0' doesn't match value '2' in RocksDBCache.obj
4>

I thought I had made it past those but do not remember how.

**Comment by gwaldron** [comment]
You might have internet access problems. Set these env vars are try `osgearth_viewer arcgisonline.earth`

set OSGEARTH_NOTIFY_LEVEL=INFO
set OSGEARTH_HTTP_DEBUG=1
